app/crud/wh_sched.py

```diff
@@ -0,0 +1,314 @@
+        or warehouse_next.scale_min == 0
+        or warehouse_now.scale_min == 0
+        or warehouse_next.scale_min == 0
+    ):
+        is_standard = True
+    else:
+        is_standard = False
+    if warehouse_now.size != warehouse_next.size:
+        changes.append(
+            f"WAREHOUSE_SIZE = {_WAREHOUSE_SIZE_OPTIONS[warehouse_next.size]}"
+        )
+        if "Snowpark" in warehouse_next.size and "Snowpark" not in warehouse_now.size:
+            changes.append("WAREHOUSE_TYPE = SNOWPARK-OPTIMIZED")
+        if "Snowpark" not in warehouse_next.size and "Snowpark" in warehouse_now.size:
+            changes.append("WAREHOUSE_TYPE = STANDARD")
+    if warehouse_now.suspend_minutes != warehouse_next.suspend_minutes:
+        changes.append(f"AUTO_SUSPEND = {warehouse_next.suspend_minutes}")
+    if warehouse_now.resume != warehouse_next.resume:
+        changes.append(f"AUTO_RESUME = {warehouse_next.resume}")
+    if warehouse_now.scale_min != warehouse_next.scale_min and not is_standard:
+        changes.append(f"MIN_CLUSTER_COUNT = {warehouse_next.scale_min}")
+    if warehouse_now.scale_max != warehouse_next.scale_max and not is_standard:
+        changes.append(f"MAX_CLUSTER_COUNT = {warehouse_next.scale_max}")
+    if (
+        warehouse_now.warehouse_mode != warehouse_next.warehouse_mode
+        and not is_standard
+    ):
+        changes.append(f"SCALING_POLICY = {warehouse_next.warehouse_mode}")
+    return ", ".join(changes)
+
+
+def update_warehouse(warehouse_now, warehouse_next):
+    set_stmts = compare_warehouses(warehouse_now, warehouse_next)
+    if set_stmts:
+        return f"alter warehouse {warehouse_next.name} set {set_stmts}"
+    return ""
+
+
+def build_task_table(session, this_run):
+    today = this_run.date()
+    # TODO handle looking back over a weekend boundary
+    # TODO handle the timestamp from config
+    is_weekday = this_run.weekday() < 5
+
+    last_run = (
+        session.table("internal.task_warehouse_schedule")
+        .filter(col("success") is True)
+        .select(sp_max("run").as_("x"))
+        .collect()[0]
+        .X
+    )
+    if last_run is None:
+        last_run = this_run - datetime.timedelta(days=1)
+    scheds = (
+        session.table("internal.wh_schedules")
+        .filter(col("weekday") == is_weekday)
+        .filter(col("enabled") is True)
+        .to_pandas()
+    )
+    if scheds.empty:
+        return scheds
+    scheds["ts"] = scheds.START_AT.map(lambda x: build_ts(today, this_run, x))
+    scheds["should_run"] = scheds.ts.map(lambda x: last_run <= x <= this_run)
+    to_run = scheds[scheds.should_run].sort_values(by="ts").groupby("NAME").first()
+
+    return to_run.reset_index()
+
+
+def build_ts(today, now, x):
+    if x < now.time():
+        return datetime.datetime.combine(today, x).replace(tzinfo=now.tzinfo)
+    else:
+        return datetime.datetime.combine(today - datetime.timedelta(days=1), x).replace(
+            tzinfo=now.tzinfo
+        )
```

Comment:
I've been going over this method, and I can't seem to figure out what exactly you were trying to do in the else-branch.

In the if-branch, for each schedule in a warehouse, you're fixing the start time to today's date and then switching the timezone. Makes sense.

In the else branch, you're the schedule(s) which have not _yet_ run to yesterday's date and then switching the timezone.

My interpretation would be that we want to always do `datetime.datetime.combine(today, x).replace(tzinfo=now.tzinfo)` and remove the `if/else` branch. The "should_run" column in the next line in the calling function would naturally filter the schedules which should not _yet_ run. What am I missing? :)